bentoml/marshal/marshal.py

```diff
@@ -197,14 +208,12 @@ def setup_routes_from_pb(self, bento_service_metadata_pb):
         for api_pb in bento_service_metadata_pb.apis:
             if api_pb.batch:
                 max_latency = (
-                    self.mb_max_latency
-                    or api_pb.mb_max_latency
-                    or self.DEFAULT_MAX_LATENCY
+                    api_pb.mb_max_latency
+                    or self.mb_max_latency
                 )
                 max_batch_size = (
-                    self.mb_max_batch_size
-                    or api_pb.mb_max_batch_size
-                    or self.DEFAULT_MAX_BATCH_SIZE
+                    api_pb.mb_max_batch_size
+                    or self.mb_max_batch_size
```

Comment:
This logic is the only backward incompatible part. CLI parameter `self.mb_max_latency` and legacy configuration key `self.DEFAULT_MAX_LATENCY` were already resolved outside of this class. Therefore we cannot apply the override priority as specified in the old logic. The new logic will prioritize the `max_latency` and `max_batch_size` specified in the Bento metadata over CLI parameter.

We should discuss our long term plan fo the metadata parameters. How they will co-exist (or not) with the configuration in the bento bundle.